scripts/scil_validate_bids.py

```diff
@@ -74,220 +104,150 @@ def _load_bidsignore_(bids_root):
-                if 'PhaseEncodingDirection' in nfmap_metadata:
-                    fmap_PE = nfmap_metadata['PhaseEncodingDirection']
-                    fmap_PE = fmap_PE.replace(fmap_PE[0], conversion[fmap_PE[0]])
-
-                    opposite_PE = PE.index(fmap_PE)
-                    if 'TotalReadoutTime' in dwi_metadata:
-                        if 'TotalReadoutTime' in nfmap_metadata:
-                            dwi_RT = dwi_metadata['TotalReadoutTime']
-                            fmap_RT = nfmap_metadata['TotalReadoutTime']
-                            if dwi_RT != fmap_RT and totalreadout == '':
-                                totalreadout = 'error_readout'
-                                topup_fmap[opposite_PE] = 'error_readout'
-                            elif dwi_RT == fmap_RT:
-                                topup_fmap[opposite_PE] = nfmap.path
-                                totalreadout = dwi_RT
-                    else:
-                        topup_fmap[opposite_PE] = nfmap.path
-                        totalreadout = default_readout
-
-        if sbref[index] != '' and len(sbref[index]) == 1:
-            topup_sbref[index] = sbref[index][0].path
-
-    if len(dwi) == 2:
-        if not any(s == '' for s in topup_sbref):
-            topup = topup_sbref
-        elif not any(s == '' for s in topup_fmap):
-            topup = topup_fmap
+
+    if len(dwis) == 2:
+        dwi_path[1] = dwis[1].path
+        bvec_path[1] = layout.get_bvec(dwis[1].path)
+        bval_path[1] = layout.get_bval(dwis[1].path)
+        if 'direction' in dwis[1].entities:
+            PE[1] = conversion[dwis[1].entities['direction']]
+        elif 'PhaseEncodingDirection' in dwis[1].entities:
+            PE[1] = conversion[dwis[1].entities['PhaseEncodingDirection']]
+
+    curr_dwi = dwis[0]
+    dwi_path[0] = curr_dwi.path
+    bvec_path[0] = layout.get_bvec(curr_dwi.path)
+    bval_path[0] = layout.get_bval(curr_dwi.path)
+
+    if 'TotalReadoutTime' in curr_dwi.entities:
+        totalreadout = curr_dwi.entities['TotalReadoutTime']
+
+    if 'session' in curr_dwi.entities:
+        nSess = curr_dwi.entities['session']
+
+    if 'run' in curr_dwi.entities:
+        nRun = curr_dwi.entities['run']
+
+    IntendedForPath = os.path.sep.join(curr_dwi.relpath.split(os.path.sep)[1:])
+    related_files = layout.get(part="mag",
+                               IntendedFor=IntendedForPath,
+                               regex_search=True,
+                               TotalReadoutTime=totalreadout,
+                               invalid_filters='drop') +\
+                    layout.get(part=Query.NONE,
+                               IntendedFor=IntendedForPath,
+                               regex_search=True,
+                               TotalReadoutTime=totalreadout,
+                               invalid_filters='drop')
+
+    related_files = [curr_related for curr_related in related_files if curr_related.suffix != 'dwi']
```

Comment:
`.suffix` is deprecated. Switch for `.entities['suffix']`. Also the line overflows